packages/hint-manifest-icons/src/hint.ts

```diff
@@ -0,0 +1,220 @@
+                const message = `Icon could not be fetched (request failed).`;
+
+                context.report(iconPath, message);
+
+                return null;
+            }
+            const response = networkData.response;
+
+            if (response.statusCode !== 200) {
+                const message = `Icon could not be fetched (status code: ${response.statusCode}).`;
+
+                context.report(iconPath, message);
+
+                return null;
+            }
+
+            return response.body.rawContent;
+        };
+        /**
+         * Passes only for the PNG files
+         * @param iconType type specified in the manifest file
+         * @param rawContent raw datastream
+         * @param iconPath icon resource path
+         */
+        const validateImageType = (iconType: string | undefined, rawContent: Buffer, iconPath: string) => {
+            const allowedTypes = ['png', 'jpg'];
+
+            if (iconType === undefined) {
+                const message = `Icon type was not specifed`;
+
+                context.report(iconPath, message);
+
+                return false;
+            }
+
+            const specifiedType = iconType.split('/')[1];
+
+            /** Handling for the corrupt rawContent */
+            if (rawContent !== null) {
+                const image: ImageTypeResult | null = imageType(rawContent);
+
+                if (image !== null) {
+                    const isValidType: boolean = allowedTypes.some((item) => {
+                        return item === image.ext;
+                    });
+
+                    if (specifiedType !== image.ext) {
+                        const message = `Real image type (${image.ext}) do not match with specified type (${specifiedType})`;
+
+                        context.report(iconPath, message);
+                    }
+
+                    if (isValidType) {
+                        return true;
+                    }
+                }
+
+                const message = `Icon should be a valid image type ${JSON.stringify(allowedTypes)}`;
+
+                context.report(iconPath, message);
+
+                return false;
+            }
+
+            return false;
+        };
+
+        /**
+         *
+         * @param iconSizes Sizes specified in the manifest file
+         * @param iconRawData
+         * @param iconPath full path to the icon file
+         */
+        const validateSizes = (iconSizes: string | undefined, iconRawData: Buffer, iconPath: string) => {
+            if (iconSizes === undefined) {
+                context.report(iconPath, `Sizes not specifed for icon`);
+
+                return false;
+            }
+            /**
+             * sizes can be the string 'any' OR
+             * two non-negative integers without leading 0s and separated by 'x' like 144x144
+             */
+            if (iconSizes === 'any') {
+                return false;
+            }
+            const specifiedSize = iconSizes.split('x');
```

Comment:
Per https://developer.mozilla.org/en-US/docs/Web/Manifest#icons this can also be a space-separated list of sizes (mostly used for the `ico` format). We should probably warn if there's more than one size specified since our supported formats will only have a single size.

Reply:
@antross  Yes, it can be a space-separated list of sizes. I was thinking of reporting in case **none** of the specified sizes match the real size. Otherwise, pass. 

Or should it strictly report if a list of sizes is encountered?

Reply:
After double-checking the spec I think your plan of reporting only if **none** of the specified sizes match the real size is best.

Per https://w3c.github.io/manifest/#sizes-member:
> When multiple ImageResources are available, a user agent MAY use the value to decide which icon is most suitable for a display context (and ignore any that are inappropriate).

I read that as suggesting a developer may need to specify additional sizes that aren't exact matches to influence where the icon gets used, so we should allow it.

Reply:
We should link to this part of the spec in the code so we don't have to look it up again.

Reply:
Added the check and the test case.